code/datums/mapgen/GrassGenerator.dm

```diff
@@ -40,7 +40,7 @@
 		var/turf/gen_turf = t
 		var/BIOME_RANDOM_SQUARE_DRIFT = 2
 		var/drift_x = (gen_turf.x + rand(-BIOME_RANDOM_SQUARE_DRIFT, BIOME_RANDOM_SQUARE_DRIFT)) / perlin_zoom
-		var/drift_y = (gen_turf.y + rand(-BIOME_RANDOM_SQUARE_DRIFT, BIOME_RANDOM_SQUARE_DRIFT)) / perlin_zoom
+		var/drift_y = (gen_turf.y + rand(0, BIOME_RANDOM_SQUARE_DRIFT)) / perlin_zoom
 
 		var/height = text2num(rustg_noise_get_at_coordinates("[height_seed]", "[drift_x]", "[drift_y]"))
```

Comment:
Could you explain what you mean by this and what the code is intended to do here?

![image](https://github.com/user-attachments/assets/d4ddf93a-84a1-4f33-b4af-65090c8ec6da)

Reply:
To move its Y-offset to a minimum of negative 0. Currently it is blocking players standing underneath them in some locations. While it looks good when they are a tile above it, it looks weird when they are standing underneath it.

Reply:
This isn't what the code is doing, you are changing the Y sampling position of the perlin noise which means that biomes won't have dithering along the Y axis, only along the X axis. This will not affect the actual y position that the trees spawn at, as that would require modifying their pixel_y values.

Noise is a 2D texture, you are changing where you are sampling the noise from rather than anything to do with the trees themselves.

Reply:
Oh dang, thank you for catching that error!!